src/generators/cfml.ts

```diff
@@ -0,0 +1,116 @@
+  let cfmlCode = "";
+
+  cfmlCode += "httpService = new http();\n";
+  cfmlCode += 'httpService.setUrl("' + quote(request.url as string) + '");\n';
+  cfmlCode += 'httpService.setMethod("' + quote(request.method) + '");\n';
+
+  if (request.cookies) {
+    for (const [headerName, headerValue] of request.cookies) {
+      cfmlCode +=
+        'httpService.addParam(type="cookie", name="' +
+        quote(headerName) +
+        '", value="' +
+        quote(headerValue) +
+        '");\n';
+    }
+    util.deleteHeader(request, "Cookie");
+  }
+
+  if (request.headers && request.headers.length) {
+    for (const [headerName, headerValue] of request.headers) {
+      cfmlCode +=
+        'httpService.addParam(type="header", name="' +
+        quote(headerName) +
+        '", value="' +
+        quote(headerValue as string) +
+        '");\n';
+    }
+  }
+
+  if (request.timeout) {
+    cfmlCode +=
+      "httpService.setTimeout(" + (parseInt(request.timeout) || 0) + ");\n";
+  }
+
+  if (request.auth) {
+    const [authUser, authPassword] = request.auth;
+    cfmlCode += 'httpService.setUsername("' + quote(authUser) + '");\n';
+    cfmlCode +=
+      'httpService.setPassword("' + quote(authPassword || "") + '");\n';
+  }
+
+  if (request.proxy) {
+    let proxy = request.proxy;
+    let proxyPort = "1080";
+    const proxyPart = (request.proxy as string).match(/:([0-9]+)/);
+    if (proxyPart) {
+      proxy = request.proxy.slice(0, proxyPart.index);
+      proxyPort = proxyPart[1];
+    }
+
+    cfmlCode += 'httpService.setProxyServer("' + quote(proxy) + '");\n';
+    cfmlCode += "httpService.setProxyPort(" + quote(proxyPort) + ");\n";
+
+    if (request.proxyAuth) {
+      const [proxyUser, proxyPassword] = request.proxyAuth.split(/:(.*)/s, 2);
+      cfmlCode += 'httpService.setProxyUser("' + quote(proxyUser) + '");\n';
+      cfmlCode +=
+        'httpService.setProxyPassword("' + quote(proxyPassword || "") + '");\n';
+    }
+  }
+
+  if (request.data || request.multipartUploads) {
+    if (request.multipartUploads) {
+      for (const [multipartKey, multipartValue] of request.multipartUploads) {
+        if (multipartValue.charAt(0) === "@") {
+          cfmlCode +=
+            'httpService.addParam(type="file", name="' +
+            quote(multipartKey) +
+            '", file="#expandPath("' +
+            quote(multipartValue.substring(1)) +
+            '")#");\n';
+        } else {
+          cfmlCode +=
+            'httpService.addParam(type="formfield", name="' +
+            quote(multipartKey) +
+            '", value="' +
+            quote(multipartValue) +
+            '");\n';
+        }
+      }
+    } else if (
+      !request.isDataRaw &&
+      (request.data as string).charAt(0) === "@"
+    ) {
+      cfmlCode +=
+        'httpService.addParam(type="body", value="#fileReadBinary(expandPath("' +
+        quote((request.data as string).substring(1)) +
+        '"))#");\n';
```

Comment:
Is there a `fileRead` as opposed to `fileReadBinary` or something? We could have two cases

```suggestion
    } else if (
      !request.isDataRaw &&
      (request.data as string).charAt(0) === "@"
    ) {
      cfmlCode +=
        'httpService.addParam(type="body", value="#' + 
        (request.isDataBinary ? "fileReadBinary" : "fileRead") +
        '(expandPath("' +
        quote((request.data as string).substring(1)) +
        '"))#");\n';
```

Reply:
There is a fileread but and it returns a string

Reply:
Sorry, the "but" in your comment is throwing me off a bit. This change makes sense, right?

Reply:
Sorry I was supposed to just say. Yes it does exist and it returns a string